### PR TITLE
cleaner start of crawler

### PR DESCRIPTION
**Pull request by flepied**
- add dependencies between services in docker-compose.yml
- test the connection before using the elastic API

### CODE REVIEW

File: monocle/db/db.py [hunk 1]
```diff
@@ -36,17 +37,26 @@ class ELmonocleDB():
...
-                self.log.info(self.es.info())
+                s.connect((ip, int(port)))
+                s.shutdown(2)
+                s.close()
                 break
-            except Exception:
-                self.log.exception('Unable to connect to %s. Sleeping.'
-                                   % elastic_conn)
+            except Exception as excpt:
+                self.log.info('Unable to connect to %s: %s. Sleeping.'
```
Comment (morucci) [inline monocle/db/db.py:52]: Why not using self.log.exception instead ?

File: monocle/db/db.py [hunk 2]
```diff
@@ -36,17 +37,26 @@ class ELmonocleDB():
...
+                              % (elastic_conn, excpt))
                 time.sleep(10)
```
Comment (morucci) [inline monocle/db/db.py:54]: Could you set the timeout in a var

File: monocle/db/db.py (hunk 1 above)
Comment (morucci) [inline monocle/db/db.py:52]: 'Sleeping for %s' % cnx_sleep_timeout

Comment (flepied) [reply monocle/db/db.py:52]: because it's too verbose

Comment (flepied) [reply monocle/db/db.py]: ok done

File: monocle/db/db.py (hunk 2 above)
Comment (flepied) [reply monocle/db/db.py]: done

File: monocle/main.py [hunk 3]
```diff
@@ -131,7 +134,8 @@ def main():
                     loop_delay=project['crawler']['loop_delay'],
                     token=crawler_item['token'],
                     base_url=crawler_item['base_url'])
-                tpool.append(Crawler(c_args, elastic_conn=args.elastic_conn))
+                tpool.append(Crawler(c_args, elastic_conn=args.elastic_conn,
+                                     elastic_timeout=args.elastic_timeout))
```
Comment (morucci) [inline monocle/main.py:138]: You need that change on the gerrit crawler too, few lines below.

File: monocle/db/db.py [hunk 4]
```diff
@@ -35,17 +36,27 @@ class ELmonocleDB():
...
+                s.connect((ip, int(port)))
+                s.shutdown(2)
+                s.close()
                 break
-            except Exception:
-                self.log.exception('Unable to connect to %s. Sleeping.'
-                                   % elastic_conn)
-                time.sleep(10)
+            except Exception as excpt:
+                self.log.info('Unable to connect to %s: %s. Sleeping for %ds.'
```
Comment (morucci) [inline monocle/db/db.py:53]: Don't you want to use self.log.exception ?

Comment (flepied) [reply monocle/db/db.py:53]: nope because it was done on purpose to avoid to scare people for nothing

File: monocle/main.py (hunk 3 above)
Comment (flepied) [reply monocle/main.py:138]: done

**Pull request merged**